### ISSUE TITLE
Rule options available as interfaces

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#1676]
_**Context**: I'm working on [autotslint](https://github.com/autolint/autotslint), a CLI tool to take in TSLint failures and, when possible, fix them in the files._

_**Ask**_: Can existing rules have interfaces for their options made available?

AFAIK there's no source of interfaces/types available in code for rules. We as humans know what they take in from the static metadata objects but that isn't available in code. They'd be very useful for autotslint, in parsing the more complex rules!

A couple of possibilities:
* I go through all existing rules and write `export type Options = ...` in all of them *(would rather not!)*.
* I write some fancy custom build logic using one of the TypeScript generation APIs that creates some sort of .d.ts file(s) with them.

One moderately complicated example is `AlignRule`:

```typescript
export type Option = "arguments" | "parameters" | "statements";
export type Options = [boolean, Option] | [boolean, Option, Option] | [boolean, Option, Option, Option];
```

**Comment by JKillian** [comment]
Yep, ideally this would have one set of data that would generate the metadata and the actual typings. I still want to validate the options people provide in their `tslint.json` file one day which the metadata version make possible.

Do you actually need compile-time validation for the rules' options? (Although I do agree it would be nice...) Possibly runtime validation is what you really want? If we implemented the `validate` function, which takes the metadata and validates that a given configuration is good, I think we could use it for validating `tslint.json` files and for validation in your case?

**Comment by JoshuaKGoldberg** [comment]
Validating `tslint.json` files would actually be really nice now that you mention it. Filed #1681 to address it.

My use case is that when writing a mutator for a specific rule it'll want to be able to understand the settings. Paraphrased example:

```typescript
// These should be imported from "tslint" instead of declared locally
type IRuleMode = "always" | "never";
type IRuleConfig = [true, IRuleMode];

class SemicolonMutator extends Mutator {
    protected visitFailure(fileContents: string, ruleFailure: IRuleFailure, config: IRuleConfig): void {
        switch (config[1]) {
            case "always":
                this.addMutation(new TextInsertMutation(ruleFailure.position.start, ";"));
                break;

            case "never":
                this.addMutation(new TextDeleteMutation(ruleFailure.position.start, ruleFailure.position.end));
                break;
        }
    }
}

```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
